### ISSUE TITLE
Gauge download performance regression

### ISSUE THREAD

**Issue by mikeaclark** [lattice/quda#207]
Versus the tifr_redux branch, the current master branch appears to have a performance regression for downloading the gauge field.  Profile output comparing the two branches running the same parameters.

master
```
loadGaugeQuda Total time = 14.9075 secs
  download     = 13.557405 secs (  90.9%), with      315 calls at 4.303938e+04 us per call

computeGaugeForceQuda Total time = 44.7811 secs
  download     = 16.552125 secs (    37%), with      600 calls at 2.758688e+04 us per call

updateGaugeFieldQuda Total time = 60.5674 secs
  download     = 39.474821 secs (  65.2%), with      600 calls at 6.579137e+04 us per call

createExtendedGaugeField Total time = 168.859 secs
  download     = 55.388569 secs (  32.8%), with     7200 calls at 7.692857e+03 us per call
```

tifr_redux
```
loadGaugeQuda Total time = 12.1561 secs
  download     = 10.939347 secs (    90%), with      315 calls at 3.472809e+04 us per call

computeGaugeForceQuda Total time = 41.0798 secs
  download     = 14.002428 secs (  34.1%), with      600 calls at 2.333738e+04 us per call

updateGaugeFieldQuda Total time = 54.8719 secs
  download     = 34.510176 secs (  62.9%), with      600 calls at 5.751696e+04 us per call
 
createExtendedGaugeField Total time = 190.833 secs
  download     = 52.135453 secs (  27.3%), with     7200 calls at 7.241035e+03 us per call
 ```

**Comment by mathiaswagner** [comment]
Just looking through issues and noted that everything seems to be slower ... I have note benchmarked the tifr branch but we should figure out what is wrong here. Did you already check with the profiler to see whether that time is lost in the actual copy or whether there is some CPU overhead in 0.7 ?

**Comment by mikeaclark** [comment]
I haven't looked into this in any detail yet.  I put this bug in as a placeholder so I didn't forget about it.

**Comment by mikeaclark** [comment]
I have since confirmed that part of this regression was due to the `qudaCreateExtendedGaugeField` function in the milc interface not being defined consistently.  This has been fixed with #260 which addresses through the addition of `qudaResidentExtendedGaugeField`.  Together with the vectorization load/store fixes in #253, performance regressions seem to have been taken care of.
